File: frontend/lib/features/parking/models/parking_lot.dart
class ParkingLot {
  final String id;
  final String name;
  final int totalSpots;
  final double ratePerMinute;
  final double? reservationRatePerMinute;
  final String openTime;
  final String address;
  final String closeTime;
  final bool isActive;
  final int type;

  bool get isOpen => type == 0;

  ParkingLot({
    required this.id,
    required this.name,
    required this.totalSpots,
    required this.ratePerMinute,
    this.reservationRatePerMinute,
    required this.openTime,
    required this.address,
    required this.closeTime,
    required this.isActive,
    required this.type,
  });

  factory ParkingLot.fromJson(Map<String, dynamic> json) => ParkingLot(
    id: json['id'],
    name: json['name'],
    address: json['address'] ?? '',
    totalSpots: json['totalSpots'] ?? 0,
    ratePerMinute: (json['ratePerMinute'] as num?)?.toDouble() ?? 0.0,
    reservationRatePerMinute: (json['reservationRatePerMinute'] as num?)?.toDouble(),
    openTime: json['openTime'] ?? '00:00:00',
    closeTime: json['closeTime'] ?? '00:00:00',
    isActive: json['isActive'] ?? true,
    type: (json['type'] as int?) ?? 0,
  );
}

File: backend/ParkingService.Domain/Entities/ParkingLot.cs
using ParkingService.Domain.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ParkingService.Domain.Entities
{
    public class ParkingLot
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Address { get; set; } = string.Empty;
        public ParkingLotType Type { get; set; } 
        public int TotalSpots { get; set; }
        public decimal RatePerMinute { get; set; }
        public decimal? ReservationRatePerMinute { get; set; }
        public TimeOnly OpenTime { get; set; }
        public TimeOnly CloseTime { get; set; }
        public bool IsActive { get; set; } = true;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Navigation properties
        public ICollection<ParkingSpot> ParkingSpots { get; set; } = new List<ParkingSpot>();
    }
}
